experimental/c-cpp/build-generator.py

```diff
@@ -0,0 +1,1486 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Auto OSS-Fuzz generator from inside OSS-Fuzz containers."""
+
+import argparse
+import json
+import os
+import shutil
+import subprocess
+from abc import abstractmethod
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import cxxfilt
+import openai
+import yaml
+
+MAX_FUZZ_PER_HEURISTIC = 15
+INTROSPECTOR_OSS_FUZZ_DIR = '/src/inspector'
+
+client = openai.OpenAI(api_key=os.getenv('OPENAI_API_KEY'))
+
+FUZZER_PRE_HEADERS = """#include <stdlib.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+"""
+
+CPP_BASE_TEMPLATE = """#include <stdint.h>
+#include <iostream>
+
+extern "C" int
+LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
+{
+    std::string input(reinterpret_cast<const char*>(data), size);
+
+    // Insert fuzzer contents here
+    // input string contains fuzz input.
+
+    // end of fuzzer contents
+
+    return 0;
+}"""
+
+
+############################################################
+#### Logic for auto building a given source code folder ####
+############################################################
+class AutoBuildContainer:
```

Comment:
Can we separate the autobuild stuff out into a separate module?

Reply:
Yes, done.